uwh-refbox/src/main.rs

```diff
@@ -482,104 +481,544 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
+        let penalty_grab_wall_button = new_toggle_button("K", &["blue"]);
+
+        let advantage_add_cancel_button = new_button("CANCEL", &["red"]);
+        let advantage_add_submit_button = new_button("SUBMIT", &["green"]);
+
+        let advantage_add_page_header = new_label("ADD ADVANTAGE DETAILS", "header-gray");
+
+        let (advantage_keypad, _advantage_player_number) = new_keypad();
+
+        let penalty_selection_grid = gtk::Grid::new();
+        penalty_selection_grid.set_column_homogeneous(true);
+        penalty_selection_grid.set_row_homogeneous(true);
+        penalty_selection_grid.set_column_spacing(BUTTON_SPACING.try_into().unwrap());
+        penalty_selection_grid.set_row_spacing(BUTTON_SPACING.try_into().unwrap());
+
+        penalty_selection_grid.attach(&penalty_illegal_adv_button, 0, 0, 1, 1);
+        penalty_selection_grid.attach(&penalty_illegal_stop_button, 1, 0, 1, 1);
+        penalty_selection_grid.attach(&penalty_stick_foul_button, 2, 0, 1, 1);
+        penalty_selection_grid.attach(&penalty_free_arm_button, 3, 0, 1, 1);
+        penalty_selection_grid.attach(&penalty_obstruction_button, 4, 0, 1, 1);
+
+        penalty_selection_grid.attach(&penalty_out_bounds_button, 0, 1, 1, 1);
+        penalty_selection_grid.attach(&penalty_illegal_sub_button, 1, 1, 1, 1);
+        penalty_selection_grid.attach(&penalty_grab_wall_button, 2, 1, 1, 1);
+        penalty_selection_grid.attach(&penalty_false_start_button, 3, 1, 1, 1);
+        penalty_selection_grid.attach(&penalty_delay_game_button, 4, 1, 1, 1);
+
+        penalty_selection_grid.attach(&penalty_unsportsman_button, 0, 2, 1, 1);
+        penalty_selection_grid.attach(&advantage_add_cancel_button, 1, 2, 2, 1);
+        penalty_selection_grid.attach(&advantage_add_submit_button, 3, 2, 2, 1);
+
+        let advantage_type_select_layout = gtk::Grid::new();
+        advantage_type_select_layout.set_column_homogeneous(true);
+        advantage_type_select_layout.set_row_homogeneous(true);
+        advantage_type_select_layout.set_column_spacing(BUTTON_SPACING.try_into().unwrap());
+        advantage_type_select_layout.set_row_spacing(BUTTON_SPACING.try_into().unwrap());
+
+        advantage_type_select_layout.attach(&advantage_add_page_header, 0, 0, 3, 1);
+        advantage_type_select_layout.attach(&advantage_w_select_button, 0, 1, 1, 2);
+        advantage_type_select_layout.attach(&advantage_equal_select_button, 1, 1, 1, 2);
+        advantage_type_select_layout.attach(&advantage_b_select_button, 2, 1, 1, 2);
+
+        advantage_add_layout.attach(&advantage_keypad, 0, 0, 4, 9);
+        advantage_add_layout.attach(&advantage_type_select_layout, 4, 0, 8, 3);
+        advantage_add_layout.attach(&penalty_selection_grid, 4, 3, 8, 6);
+
+
+        // Setting up the white/equal/black selected buttons
+        advantage_w_select_button.connect_clicked(clone!(@strong advantage_equal_select_button, @strong advantage_b_select_button => move |b| {
```

Comment:
These buttons and the "type of foul" buttons need an initial value. For each set, should choose one and `.set_active(true)` it right after the `.connect_clicked()`s